### PR TITLE
Adds single date filter to user aggregate status report

### PR DESCRIPTION
**Pull request by ajeety4**
## Product Description

- Adds single date filter to user aggregate status report. This allow users to select the date from which they wish to see the results. 
- By default, the selected date is 60 days in past including today. User can select any date from yesterday to 1 year in the past.
- The summary metrics `in the last n days` at the top of graph is displayed dynamically (applicable ones) based on period selected by user. 
- adds horizontal scroll for large datasets for readabilty

image

## Technical Summary

https://dimagi.atlassian.net/browse/SC-3413

### Safety story

There is no change in core logic except the period is now selected by user instead of 60 days hardcoded.
Local and staging testing done.

### QA Plan

[QA ticket](https://dimagi.atlassian.net/browse/QA-6191) 

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/static/reports/js/aggregate_user_status.js
Comment (zandre-eng) [inline corehq/apps/reports/static/reports/js/aggregate_user_status.js:36]: How was the number 120 chosen (was it through visually testing the report)? Also, how does this constant hold up when a user has different browser zoom levels?

File: corehq/apps/reports/static/reports/js/aggregate_user_status.js
Comment (zandre-eng) [inline corehq/apps/reports/static/reports/js/aggregate_user_status.js:37]: Nit: Could we put the magic number inside a variable?

**Review by zandre-eng — COMMENTED** [review]
Overall looks good. Just a few minor comments/questions from my end.

File: corehq/apps/reports/standard/deployments.py
Comment (ajeety4) [reply corehq/apps/reports/standard/deployments.py:653]: We do have one generic single date filter in `SingleDateFilter` in `filters/dates.py`. 
Included this here as it (attributes) is very specific to this report and do not see scope of reuse.

File: corehq/apps/reports/static/reports/js/aggregate_user_status.js
Comment (ajeety4) [reply corehq/apps/reports/static/reports/js/aggregate_user_status.js:36]: Correct. The number was chosen such that graph is readable visually for days greater than 120. This is applicable for chart width multiplication factor (15) too. 

> Also, how does this constant hold up when a user has different browser zoom levels?

Good question here. So for period with 120 or less, the graph (bar size) automatically adjusts based on zoom.
For other case, the graph just gets zoomed out or zoom in based on browser zoom levels.

File: corehq/apps/reports/standard/deployments.py
Comment (zandre-eng) [reply corehq/apps/reports/standard/deployments.py:653]: That makes sense. Sounds good to leave as is.

**Comment by Charl1996** [comment]
Thanks for this! 

Looks good from my side, so I'm happy to approve once the linting issues is fixed. :+1:

**Review by Charl1996 — APPROVED** [review]

**Review by zandre-eng — APPROVED** [review]
Thanks for the feedback on my comments @ajeety4. Looks good from my side!

**Comment by ajeety4** [comment]
Hi @zandre-eng @Charl1996 ,
Requesting for your re approval. Wording changes done based on AE feedback in https://github.com/dimagi/commcare-hq/pull/34253/commits/89da5d9b2fbf331cd197f688f8d73860c535bedb

Thank you.

**Review by zandre-eng — APPROVED** [review]
:heavy_check_mark:

**Review by Charl1996 — APPROVED** [review]

**Review by zandre-eng — APPROVED** [review]

**Pull request merged**
